src/bp/core/routers/telemetry.ts

```diff
@@ -0,0 +1,56 @@
+import * as sdk from 'botpress/sdk'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import AuthService, { TOKEN_AUDIENCE } from 'core/services/auth/auth-service'
+import { RequestHandler, Router } from 'express'
+
+import { CustomRouter } from './customRouter'
+import { checkTokenHeader } from './util'
+
+export class TelemetryRouter extends CustomRouter {
+  private checkTokenHeader: RequestHandler
+
+  constructor(
+    private logger: sdk.Logger,
+    private authService: AuthService,
+    private telemetryRepo: TelemetryRepository
+  ) {
+    super('Telemetry', logger, Router({ mergeParams: true }))
+    this.checkTokenHeader = checkTokenHeader(this.authService, TOKEN_AUDIENCE)
+    this.setupRoutes()
+  }
+
+  private setupRoutes() {
+    // TODO get server header
+    this.router.post(
+      '/feedback',
+      this.checkTokenHeader,
+      this.asyncMiddleware(async (req, res) => {
+        const { success, events } = req.body
+
+        if (!events.length || !!events.filter(e => typeof e !== 'string').length) {
```

Comment:
please use joi for validation and please in case of failure send a 400

Reply:
Fixed in e7e4566 & e47a42f (small addition)!